cmd/integration-test-server/main.go

```diff
@@ -278,6 +288,202 @@ func downloadBlobFromRPC(client *rpc.Client, kvIndex uint64, hash common.Hash) (
 	return result, nil
 }
 
+func checkByDesignFailure() int {
+	count0, err := checkDiffNotMatchError()
+	if err != nil {
+		addErrorMessage(fmt.Sprintf("checkDiffNotMatchError fail: err, %s", err.Error()))
+	}
+	count1, err := checkInvalidSamplesError()
+	if err != nil {
+		addErrorMessage(fmt.Sprintf("checkInvalidSamplesError fail: err, %s", err.Error()))
+	}
+
+	return count0 + count1
+}
+
+func checkDiffNotMatchError() (int, error) {
+	// Description: 1. got two mining results with the same block number and diff nonce; 2. two mining results, one successful and one failed.
+	// Sample:
+	// lvl=info msg="Set mining result"         shard=1 block=6,906,682 nonce=565,740
+	// lvl=info msg="Set mining result"         shard=1 block=6,906,682 nonce=779,947
+	// lvl=eror msg="Failed to submit mined result"         shard=1 block=6,906,682 error="failed to estimate gas: execution reverted: StorageContract: diff not match"
```

Comment:
The logic should be: check the "Set mining result" timestamp of the block mined, if any mining tx confirmed since then, the "diff not match" error should be omitted.